proxy/proxy.go

```diff
@@ -717,6 +718,16 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 					err:  err,
 					code: http.StatusServiceUnavailable,
 				}
+			} else if !nerr.Timeout() && nerr.Temporary() {
+				ext.HTTPStatusCode.Set(ctx.proxySpan, uint16(http.StatusServiceUnavailable))
+				if strings.Contains(err.Error(), "read: connection reset by peer") {
+					// retrieable case https://github.com/zalando/skipper/issues/768
+					return nil, &proxyError{
+						err:           err,
+						code:          http.StatusServiceUnavailable,
+						dialingFailed: true,
```

Comment:
Does this trigger a retry?

The application never received the request so this seems just like a better error code.

Reply:
it retries only if it is a "loadbalanced" route, with multiple available endpoints. Applies to Kubernetes ingress scenarios.

Reply:
@aryszka got it, that is the case here so all good.

Reply:
I had to fix the fallback problem, because we would not return an error if it would bre no timeout and temporary without connection reset by peer...